firmware/common/external_app.hpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2023 Bernd Herzog
+ *
+ * This file is part of PortaPack.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#ifndef __EXTERNAL_APPS_H__
+#define __EXTERNAL_APPS_H__
+
+#include "ch.h"
+#include "ui_navigation.hpp"
+#include "spi_image.hpp"
+
+typedef void (*externalAppEntry_t)(ui::NavigationView& nav);
+
+enum app_location_t : uint32_t {
+    UTILITIES = 0,
+    RX,
+    TX
+};
+
+typedef struct {
```

Comment:
nit: don't need typedefs in C++.

Reply:
True. That started out to be plain C but got refactored until it was here. Will change that